srcs/app/config/analyser/ParserStatelessSet.cpp

```diff
@@ -0,0 +1,303 @@
+	|| (ctx == Parser::State::K_LOCATION && (kw == Parser::State::K_LISTEN
+									|| kw == Parser::State::K_SERVER_NAME)))
+		return false;
+	return true;
+}
+
+t_parsing_state Parser::StatelessSet::ExpKwHandlerKw(const StatefulSet &data) {
+	if (data.GetState() < Parser::State::K_SERVER
+	|| data.GetState() > Parser::State::K_LIMIT_EXCEPT)
+		throw SyntaxError("Expecting keyword but found `" +
+		data.GetRawData() + "'", LINE);
+	if (!isKwAllowedInCtx_(data.GetState(), data.GetCtx()))
+		throw SyntaxError("Keyword `" + data.GetRawData() + "' "
+						  "not allowed in context `" +
+						  Parser::State::GetParsingStateTypeStr(data.GetCtx())
+						  + "'", LINE);
+	return data.GetState();
+}
+
+t_parsing_state Parser::StatelessSet::AutoindexHandler
+													(const StatefulSet &data) {
+	if (data.GetRawData() != "on" && data.GetRawData() != "off")
+		throw SyntaxError("Expecting `on' or `off' but found `" +
+		data.GetRawData()  + "'", LINE);
+	config_->AddAutoindex(data.GetRawData() == "on", data.GetCtx(), LINE);
+	return Parser::State::K_EXP_SEMIC;
+}
+
+t_parsing_state Parser::StatelessSet::ServerNameHandlerSemic
+(const StatefulSet &data) {
+	if (data.GetArgNumber() == 0)
+		throw Analyser::SyntaxError("Invalid number of arguments in "
+									"`server_name' directive", LINE);
+	config_->AddServerName(parser_->GetArgs(), data.GetCtx(), LINE);
+	parser_->ResetArgNumber();
+	return Parser::State::K_EXP_KW;
+}
+
+t_parsing_state Parser::StatelessSet::ServerNameHandler
+													(const StatefulSet &data) {
+	parser_->IncrementArgNumber(data.GetRawData());
+	return Parser::State::K_SERVER_NAME;
+}
+
+bool Parser::StatelessSet::areHttpMethodsValid_(const std::vector<std::string>
+										&input, std::string *error_throw) {
+	for (unsigned int i = 0; i < input.size(); ++i) {
+		bool notValidMethod = true;
+		for (unsigned j = 0;
+			 j < sizeof(valid_http_methods)/sizeof(valid_http_methods[0]);
+			 ++j )
+			notValidMethod = notValidMethod
+				&& (valid_http_methods[j] != input[i]);
+		if (notValidMethod) {
+			*error_throw = "`" + input[i] + "' is not"
+				" a valid http method for `limit_except' directive";
+			return false;
+		}
+	}
+	return true;
+}
+
+
+t_parsing_state Parser::StatelessSet::LimitExceptHandlerSemic
+(const StatefulSet &data) {
+	std::string error_throw;
+	if (data.GetArgNumber() == 0)
+		throw Analyser::SyntaxError("Invalid number of arguments in "
+									"`limit_except' directive", LINE);
+	if (!areHttpMethodsValid_(parser_->GetArgs(), &error_throw)) {
+		throw Analyser::SyntaxError(error_throw, LINE);
+	}
+	config_->AddLimitExcept(parser_->GetArgs(), data.GetCtx(), LINE);
+	parser_->ResetArgNumber();
+	return Parser::State::K_EXP_KW;
+}
+
+
+t_parsing_state Parser::StatelessSet::LimitExceptHandler
+(const StatefulSet &data) {
+	parser_->IncrementArgNumber(data.GetRawData());
+	return Parser::State::K_LIMIT_EXCEPT;
+}
+
+t_parsing_state Parser::StatelessSet::CgiAssignHandler
```

Comment:
Here we could improve the extension validation. For example, an extension should start with a `.` and have at least one more character after.

Also, can we add the extension to the map without the `.`, please? 
This way, it will integrate seamlessly with the HTTP part.
Example: `.py -> py`

Reply:
I was trying your CGI testing environment in the other branch I have, and the cgi was not working. I was wondering why, and this is it. Now it should be okay. Thx!